src/intrinsics/ecma262/Object.js

```diff
@@ -274,17 +274,20 @@ export default function(realm: Realm): NativeFunctionValue {
         to.makeSimple();
 
         // Tell serializer that it may add properties to to only after temporalTo has been emitted
-        let temporalArgs = [ObjectAssign, to, ...delayedSources];
+        let temporalArgs = [to, ...delayedSources];
+        let preludeGenerator = realm.preludeGenerator;
+        invariant(preludeGenerator !== undefined);
         let temporalTo = AbstractValue.createTemporalFromBuildFunction(
```

Comment:
This function call with the `Object.assign` build node is a lot of copy&paste that appears 5 times in the code. I think this calls for a dedicated `AbstractValue.createTemporalObjectAssign` function.

Reply:
I'll do in a follow up PR, if that's okay?